### ISSUE TITLE
otp 24.0.6 demonitor/3 aliased ref core dumped

### ISSUE THREAD

**Issue by hizel** [erlang/otp#5225]
**Describe the bug**
demonitor aliased ref results as unexpected error and core dump

**To Reproduce**
```erlang
$erl
Erlang/OTP 24 [erts-12.0.4] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit] [lttng]

Eshell V12.0.4  (abort with ^G)
1> Pid = spawn(fun F() -> timer:sleep(10), F() end).
<0.82.0>
2> Ref1 = monitor(process, Pid, [{alias, explicit_unalias}]).
#Ref<0.1672086153.3663003650.98647>
3> demonitor(Ref1).
true
4> exit(Pid, stop).
true
5> is_process_alive(Pid).
false
6> Ref2 = monitor(process, Pid, [{alias, explicit_unalias}]).
#Ref<0.1672086153.3663003650.98664>
7> demonitor(Ref2).                                          
beam/bif.c:473:demonitor(): Internal error: Unexpected monitor type
Aborted (core dumped)
```

**Expected behavior**
Expect result true|false or erlang exception .

**Affected versions**
OTP 24.0.6, 24.0.2 x86_64

**Comment by rickard-green** [comment]
PR #5230 (based on OTP 24.1) should fix this issue. Note that it has not passed more than some very basic testing yet.

**Comment by rickard-green** [comment]
BTW Thanks for yet another detailed bug report making it easy to track down the bug!

**Comment by hizel** [comment]
works for me, thanks!

**Comment by rickard-green** [comment]
Now fixed in the just released [OTP 24.1.1](https://erlang.org/download/OTP-24.1.1.README) patch.
